### PR TITLE
Fix augment methods in `VectorizedBaseImageAugmentationLayer`

### PR DESCRIPTION
**Pull request by james77777778**
# What does this PR do?

I have originally reported a bug in https://github.com/keras-team/keras-cv/pull/1555#issuecomment-1477423343

This PR fixes the arguments in `_batch_augment` and the missing `**kwargs` in `augment_targets`.

This PR also adds `labels`, `bounding_boxes` and `images` for `augment_segmentation_masks` to align to `augment_keypoints`. It should be a harmless addition.

An assertion test including VectorizedAssertionLayer is added to verify the changes.

As a result, the workaround in Mosaic must be removed.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [X] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?
@LukeWood @ianstenbit

### CODE REVIEW

**Review by ianstenbit — APPROVED** [review]
Thanks for the PR, and for improving our unit test coverage!
